sys/include/net/gnrc/rpl/structs.h

```diff
@@ -238,7 +238,7 @@ struct gnrc_rpl_parent {
 typedef struct {
     uint16_t ocp;   /**< objective code point */
     uint16_t (*calc_rank)(gnrc_rpl_parent_t *parent, uint16_t base_rank); /**< calculate the rank */
-    gnrc_rpl_parent_t *(*which_parent)(gnrc_rpl_parent_t *, gnrc_rpl_parent_t *); /**< compare for parents */
+    int (*which_parent)(gnrc_rpl_parent_t *, gnrc_rpl_parent_t *); /**< compare for parents */
```

Comment:
instead of changing an existing functions (hence the API) wouldn't suffice to add a separate (helper) function, i.e. `int parent_cmp(a, b)` - which then could also be used by `which_parent` btw.